src/extractor/extraction_containers.cpp

```diff
@@ -292,23 +322,23 @@ void ExtractionContainers::PrepareEdges(ScriptingEnvironment &scripting_environm
     std::cout << "[extractor] Setting start coords      ... " << std::flush;
     TIMER_START(set_start_coords);
     // Traverse list of edges and nodes in parallel and set start coord
-    auto node_iterator = all_nodes_list.begin();
+    auto node_iterator = used_node_id_list.begin();
     auto edge_iterator = all_edges_list.begin();
 
     const auto all_edges_list_end = all_edges_list.end();
-    const auto all_nodes_list_end = all_nodes_list.end();
+    const auto used_node_id_list_end = used_node_id_list.end();
 
-    while (edge_iterator != all_edges_list_end && node_iterator != all_nodes_list_end)
+    while (edge_iterator != all_edges_list_end && node_iterator != used_node_id_list_end)
```

Comment:
I'm pretty sure this logic is incorrect, because edge creation hangs for larger graphs.  Still needs some work.